test/util/unit_test.cpp

```diff
@@ -168,6 +168,67 @@ class XmlReporter : public Reporter {
     std::ostream& m_out;
 };
 
+class JUnitReporter : public XmlReporter
+{
+public:
+    JUnitReporter(std::ostream& out)
+    : XmlReporter(out)
+    {
+    }
+
+    ~JUnitReporter() noexcept
+    {
+    }
+    void summary(const SharedContext& context, const Summary& results_summary) override
+    {
+        m_out << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n";
+
+        m_out << "<testsuites>\n"
+              << "  <testsuite "
+              << "name=\"realm-core-tests\" "
+              << "tests=\"" << results_summary.num_executed_tests << "\" "
+              << "disabled=\"" << results_summary.num_excluded_tests << "\" "
+              << "failures=\"" << results_summary.num_failed_tests << "\" "
+              << "id=\"0\" "
+              << "time=\"" << results_summary.elapsed_seconds << "\""
+              << ">\n";
+
+        std::ostringstream out;
+        out.imbue(std::locale::classic());
```

Comment:
classic() gives valgrind race errors, you can remove the imbue() call completely because because it's only needed in end-users' devices which can have arbitrary setup. In jenkins, etc, it doesn't matter